### PR TITLE
backport v5.0.3 PR #2614 and #2615 for 5.0.2 patch 1

### PR DESCRIPTION
**Pull request by sjpadgett**
- patch 1
- testing well

### CODE REVIEW

**Comment by sjpadgett** [comment]
Hi @bradymiller How does this look? Want me to bring anything else back? I can go ahead and start a patch archive if you want!

**Comment by sjpadgett** [comment]
What's up with travis 7.3. I am not catching this stuff from IDE or code sniffer! Mainly php tags.

**Comment by sjpadgett** [comment]
Thought I got latest code sniffer but maybe not. I also want to add a fix for billing manager so i'll do in 503 and cherry pick with the others I guess.

**Comment by bradymiller** [comment]
hi @sjpadgett ,
I just realized that we should put the database upgrade stuff in both patch.sql and 5_0_1-to-5_0_2_upgrade.sql since a user may use a more recent package (if we update the packages for most recent patch) for an upgrade. Sorry about that. I'll also update the following wiki page for that: https://www.open-emr.org/wiki/index.php/Steps_for_a_patch_release

**Comment by sjpadgett** [comment]
Okay but upgrade runs patch doesn't it?

File: interface/super/edit_layout.php [hunk 1]
```diff
@@ -114,7 +116,7 @@ function genGroupSelector($name, $layout_id, $default = '')
         array($layout_id)
     );
     $s  = "<select class='form-control' name='" . xla($name) . "'>";
-    $s .= "<option value=''>" . xlt('None') . "</option>";
+    $s .= "<option value=''>" . xlt('None{{Group}}') . "</option>";
```
Comment (bradymiller) [inline interface/super/edit_layout.php]: None of these labels should go into the patch or they will break translations for those terms. All that stuff will need to be in 5.0.3 to allow time for translations to get to the spreadsheet and get translated.

**Comment by bradymiller** [comment]
Only sql_patch uses patch.sql (upgrade does not run it). Kind of odd that it took 7 years to figure that out :)

I noted a couple contants that need reverting in edit_layout.php. Otherwise looks good to me.

Do you want me bring in the 4 planned commit into rel-502 first and then you bring in this PR after or vice versa?

**Comment by sjpadgett** [comment]
from sql_upgrade.php L-96
if ((!empty($v_realpatch)) && ($v_realpatch != "") && ($v_realpatch > 0)) {
        // This release contains a patch file, so process it.
        upgradeFromSqlFile('patch.sql');
    }

I also have a bug fix for billing manager so I think I will include in this PR then commit new PR for 5.0.3. So if you want go ahead and bring the 4 from 503 back then i'll finish this and bring in...

**Comment by bradymiller** [comment]
dang, I don't remember doing that. senility is catching up to me :)
Thanks for digging that up. So just ignore my 5_0_1-to-5_0_2_upgrade.sql rec.

**Comment by bradymiller** [comment]
just brought in the 4 commits into rel-502

**Comment by sjpadgett** [comment]
is it best to merge them into my branch or can I just commit as is?

**Comment by bradymiller** [comment]
Just commit as is. Looks like one of the commits I brought in did the login.php changes also which is causing a conflict with your PR. Would fix that conflict and then can bring this PR in (bring in as 1 commit).

**Comment by sjpadgett** [comment]
okay

**Comment by sjpadgett** [comment]
This is cool new billing manager validate: I bet my billing buddy @stephenwaite will like!
image
Ya can now view logs while verifying the batch:
image

**Comment by sjpadgett** [comment]
Bringing in after travis. 
I'll get a patch archive going we can add to as we go. I'll post it here so you'll have a copy...
Did you say for me to just do merge so we can practice putting the commits together? ;)

**Pull request merged**